program.py

```diff
@@ -120,53 +86,66 @@ def getToken():
+    endpoint["token"] = __token
+    
     return __token
 
 
 # ************************************************************************
-# Helper function: REQUIRED: wrapper function for sending an HTTPS message
+# REQUIRED: wrapper function for sending an HTTP message
 # ************************************************************************
 
-# Define a helper function to allow easily sending web request messages;
-# this function can later be customized to allow you to port this script to other languages.
-# All it does is take in a data object and a message type, and it sends an HTTPS
-# request to the target OMF endpoint
 
-def send_omf_message_to_endpoint(message_type, message_omf_json, action='create'):
-    # Sends the request out to the preconfigured endpoint..
+def send_message_to_omf_endpoint(endpoint, message_type, message_omf_json, action='create'):
+    '''Sends the request out to the preconfigured endpoint'''
+    global endpoint_types
 
-    global producerToken, omfEndPoint, omfVersion, sendingToOCS, username, password
     # Compress json omf payload, if specified
     compression = 'none'
-    if USE_COMPRESSION:
+    if endpoint["use-compression"]:
         msg_body = gzip.compress(bytes(json.dumps(message_omf_json), 'utf-8'))
         compression = 'gzip'
     else:
         msg_body = json.dumps(message_omf_json)
 
-    msg_headers = getHeaders(compression, message_type, action)
+    # Collect the message headers
+    msg_headers = get_headers(endpoint, compression, message_type, action)
+
+    # Send message to OMF endpoint
+    endpoints_type = endpoint["endpoint-type"]
     response = {}
-    # Assemble headers
-    if sendingToOCS:
+    # If the endpoint is OCS
+    if endpoints_type == endpoint_types[0]:
         response = requests.post(
-            omfEndPoint,
+            endpoint["omf-endpoint"],
             headers=msg_headers,
             data=msg_body,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS
+            verify=endpoint["verify-ssl"],
+            timeout=endpoint["web-request-timeout-seconds"]
         )
-    else:
+    # If the endpoint is EDS
+    elif endpoints_type == endpoint_types[1]:
         response = requests.post(
-            omfEndPoint,
+            endpoint["omf-endpoint"],
             headers=msg_headers,
             data=msg_body,
-            verify=VERIFY_SSL,
-            timeout=WEB_REQUEST_TIMEOUT_SECONDS,
-            auth=(username, password)
+            timeout=endpoint["web-request-timeout-seconds"]
+        )
+    # If the endpoint is PI
+    elif endpoints_type == endpoint_types[2]:
+        response = requests.post(
+            endpoint["omf-endpoint"],
+            headers=msg_headers,
+            data=msg_body,
+            verify=endpoint["verify-ssl"],
+            timeout=endpoint["web-request-timeout-seconds"],
+            auth=(endpoint["username"], endpoint["password"])
         )
 
```

Comment:
can get into a bad spot where someone is outside the enum

Reply:
@derekendres I am not sure what you mean by this. The verify-ssl and web-request-timeout-seconds parameters are populated automatically if they are not present when reading the configuration. The auth portion always has to be specified.